### PR TITLE
Add pod membership lock

### PR DESCRIPTION
**Pull request by thomas-waite**
# Summary
Integrates a new pod controller version which adds a pod membership lock toggle. Pod admins are now able to lock and unlock pod membership transfers

### CODE REVIEW

File: contracts/pods/PodAdminGateway.sol [hunk 1]
```diff
@@ -190,6 +209,41 @@ contract PodAdminGateway is CoreRef, IPodAdminGateway {
         }
     }
 
+    /// @notice Admin functionality to turn off pod membership transfer
+    /// @dev Permissioned to GOVERNOR, POD_ADMIN, GUARDIAN and the specific role
+    function lockMembershipTransfers(uint256 _podId)
```
Comment (ElliotFriedman) [inline contracts/pods/PodAdminGateway.sol:214]: Are all membership transfers locked by default?

Comment (thomas-waite) [reply contracts/pods/PodAdminGateway.sol:214]: Not by default, this switch needs to be toggled on by the admin and at the moment requires it the pod operator creating another pod to disable transfers. Will look at disabling by default

File: contracts/pods/PodFactory.sol [hunk 2]
```diff
@@ -116,7 +119,17 @@ contract PodFactory is CoreRef, IPodFactory {
         override
         returns (address)
     {
-        return IControllerV1(podController).podAdmin(podId);
+        return ControllerV1(podController).podAdmin(podId);
```
Comment (Joeysantoro) [inline contracts/pods/PodFactory.sol:122]: nit: no need to cast it is already casted
```suggestion
        return podController.podAdmin(podId);
```

File: contracts/pods/PodFactory.sol [hunk 3]
```diff
@@ -116,7 +119,17 @@ contract PodFactory is CoreRef, IPodFactory {
...
+    }
+
+    /// @notice Get whether membership transfers are enabled for a pod
+    function getIsMembershipTransferLocked(uint256 podId)
+        external
+        view
+        override
+        returns (bool)
+    {
+        return ControllerV1(podController).isTransferLocked(podId);
```
Comment (Joeysantoro) [inline contracts/pods/PodFactory.sol:132]:
```suggestion
        return podController.isTransferLocked(podId);
```

File: contracts/pods/PodAdminGateway.sol [hunk 4]
```diff
@@ -65,6 +68,22 @@ contract PodAdminGateway is CoreRef, IPodAdminGateway {
             );
     }
 
+    /// @notice Calculate the specific pod membership transfer lock role
```
Comment (Joeysantoro) [inline contracts/pods/PodAdminGateway.sol:71]: what does this do

File: contracts/pods/PodAdminGateway.sol (hunk 1 above)
Comment (Joeysantoro) [reply contracts/pods/PodAdminGateway.sol:214]: agree it should be disabled by default. The factory can have the role and atomically disable

**Review by Joeysantoro — COMMENTED** [review]
LGTM overall

File: contracts/pods/PodFactory.sol (hunk 2 above)
Comment (thomas-waite) [reply contracts/pods/PodFactory.sol:122]: Ah, slipped through. Corrected

File: contracts/pods/PodAdminGateway.sol (hunk 1 above)
Comment (thomas-waite) [reply contracts/pods/PodAdminGateway.sol:214]: Yep agreed

**Pull request merged**